data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/PublishedNewsDisplayedPropUpgrade.java

```diff
@@ -103,25 +121,27 @@ public void processUpgrade(String oldVersion, String newVersion) {
 
         News news = newsService.getNewsById(newsTargetsMetadataItem.getObjectId(), false);
         boolean displayed = !news.isArchived() && !StringUtils.equals(news.getPublicationState(), STAGED_STATUS);
-        StringBuilder insertNewsTargetsMetadataItemsPropsQueryStringBuilder = new StringBuilder("INSERT INTO SOC_METADATA_ITEMS_PROPERTIES(METADATA_ITEM_ID, NAME, VALUE) VALUES('");
-        insertNewsTargetsMetadataItemsPropsQueryStringBuilder.append(newsTargetsMetadataItem.getId() + "', '");
-        insertNewsTargetsMetadataItemsPropsQueryStringBuilder.append(NewsUtils.DISPLAYED_STATUS + "', '");
-        insertNewsTargetsMetadataItemsPropsQueryStringBuilder.append(displayed + "')");
-        Query insertNewsTargetsMetadataItemsPropsQuery = entityManager.createNativeQuery(insertNewsTargetsMetadataItemsPropsQueryStringBuilder.toString(), MetadataItemEntity.class);
-        insertNewsTargetsMetadataItemsPropsQuery.executeUpdate();
-        publishedNewsDisplayedPropCount++;
-      }
-      if (transactionStarted && entityManager.getTransaction().isActive()) {
-        entityManager.getTransaction().commit();
-        entityManager.flush();
-      }
-    } catch (Exception e) {
-      if (transactionStarted && entityManager.getTransaction().isActive() && entityManager.getTransaction().getRollbackOnly()) {
-        entityManager.getTransaction().rollback();
+        StringBuilder insertNewsTargetsMetadataItemsPropQueryStringBuilder = new StringBuilder("INSERT INTO SOC_METADATA_ITEMS_PROPERTIES(METADATA_ITEM_ID, NAME, VALUE) VALUES('");
```

Comment:
use `setParameter` to set parameter values of SQL query instead of concatenating value (Best practice to avoid SQL injection)

Reply:
```suggestion
                Query insertNewsTargetMetadataItemsPropQuery = entityManager.createNativeQuery("INSERT INTO SOC_METADATA_ITEMS_PROPERTIES(METADATA_ITEM_ID, NAME, VALUE) VALUES(':newsTargetsMetadataItemId', ':displatedStatus', ':displayed')", MetadataItemEntity.class);
        insertNewsTargetMetadataItemsPropQuery.setParameter("newsTargetsMetadataItemId", newsTargetsMetadataItem.getId());
        insertNewsTargetMetadataItemsPropQuery.setParameter("dispalyedStatus", NewsUtils.DISPLAYED_STATUS);
        insertNewsTargetMetadataItemsPropQuery.setParameter("dispalyed", displayed);
        insertNewsTargetMetadataItemsPropQuery.executeUpdate();
```
I have tried this syntax but no result returned by the query

Reply:
need to be tested without quotes in parameters list